app/javascript/bundles/RestaurantsApp/components/AddRestaurant.jsx

```diff
@@ -51,6 +53,7 @@ export default class AddRestaurantForm extends React.Component {
     submit(model) {
         this.setState({isFormSubmitted: true});
         if (this.state.isFormValid) {
+            this.setState({isAddingRestaurant: true});
```

Comment:
For something like this I'd consider a more generic name, like `isSubmitting`.

It would make more sense if you ever want to extract your form handling code to a separate component.